### ISSUE TITLE
Cannot install packages, just installed ST3 build 3124

### ISSUE THREAD

**Issue by hobbyman** [wbond/package_control#1163]
Our network traffic goes through a device that requires a certificate to be install on our computers.

I have the .pem files for the 2 certs that are required.

I had previously been able to install packages by pasting the contents of those 2 files into:
Package Control.merged-ca-bundle
(which I now see in the docs is the wrong place, should be in .user-ca-bundle)

Now, after upgrading to the latest build of ST3 (3124, release 9/22/16), I cannot install packages no matter where I copy the cert information.

Here is the debug information from the console:

```
Package Control: Fetching list of available packages and dependencies
  Platform: osx-x64
  Sublime Text Version: 3124
  Package Control Version: 3.2.0
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.33.29
  Resolved IPv6: 2600:3c02::f03c:91ff:fedf:6883
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/myuid/Library/Application Support/Sublime Text 3/Packages/User/oscrypto-ca-bundle.crt (311734 bytes)
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/myuid/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
```

**Comment by wbond** [comment]
Do you see your certificates in the merged bundle?

**Comment by hobbyman** [comment]
Yes, I do.

**Comment by wbond** [comment]
According to the source, the handshake is failing with `CERTIFICATE_VERIFY_FAILED`: https://hg.python.org/cpython/file/3.3/Modules/_ssl.c#l548.

I don't know why this is, but there is #1131 where I hope to add some functionality using the oscrypto library to debug TLS handshakes. Unfortunately the build-in Python `ssl` module doesn't seem to provide such functionality.

**Comment by hobbyman** [comment]
Ok, and since I can no longer install packages I will guess that "Package Control" updates won't happen either. Anyways, this isn't the fault of the programmer, the fault lies in our crazy network environment. I may go down the route of installing ST3 and Package Control fresh but I'm thinking that won't matter. Thanks.

**Comment by Shababsoft** [comment]
I am also facing same issue and getting the below error:
`Exception in thread Thread-6:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "package_control.commands.install_package_command in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 45, in run
  File "package_control.package_installer in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 58, in make_package_list
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 687, in list_available_packages
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 551, in _list_available
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 450, in list_repositories
  File "package_control.providers.channel_provider in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 201, in get_repositories
  File "package_control.providers.channel_provider in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 87, in fetch
  File "package_control.download_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 340, in fetch
  File "package_control.downloaders.wininet_downloader in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 338, in download
OSError: exception: access violation reading 0x0000000000007530`

**Comment by Shababsoft** [comment]
I am also facing same the issue and getting the below error:

`Exception in thread Thread-6:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "package_control.commands.install_package_command in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 45, in run
  File "package_control.package_installer in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 58, in make_package_list
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 687, in list_available_packages
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 551, in _list_available
  File "package_control.package_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 450, in list_repositories
  File "package_control.providers.channel_provider in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 201, in get_repositories
  File "package_control.providers.channel_provider in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 87, in fetch
  File "package_control.download_manager in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 340, in fetch
  File "package_control.downloaders.wininet_downloader in C:\Users\XXXXXX\AppData\Roaming\Sublime Text 3\Installed Packages\Package Control.sublime-package", line 338, in download
OSError: exception: access violation reading 0x0000000000007530`

**Comment by wbond** [comment]
@Shababsoft That error is 100% different than the OP. Your bug seems to be the same as #1159

**Comment by Shababsoft** [comment]
@wbond Thanks

**Comment by seripap** [comment]
Broke here as well, OSX 10.12 on build 3124. A reinstall of package control fixed it.

```
Exception in thread Thread-22:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "package_control.commands.install_package_command in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 45, in run
  File "package_control.package_installer in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 58, in make_package_list
  File "package_control.package_manager in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 687, in list_available_packages
  File "package_control.package_manager in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 551, in _list_available
  File "package_control.package_manager in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 450, in list_repositories
  File "package_control.providers.channel_provider in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 201, in get_repositories
  File "package_control.providers.channel_provider in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 87, in fetch
  File "package_control.download_manager in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 340, in fetch
  File "package_control.downloaders.urllib_downloader in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 117, in download
  File "package_control.downloaders.urllib_downloader in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 317, in setup_opener
  File "package_control.ca_certs in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 31, in get_ca_bundle_path
  File "package_control.ca_certs in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 140, in get_system_ca_bundle_path
  File "package_control.deps.oscrypto.trust_list in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 115, in get_path
  File "package_control.deps.oscrypto._osx.trust_list in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 86, in extract_from_system
  File "package_control.deps.oscrypto._osx._security in /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 50, in handle_sec_error
OSError: The Trust Settings Record was corrupted.
```

**Comment by hobbyman** [comment]
Reinstall of package control **_did not_** work for me.

**Comment by wbond** [comment]
@seripap That bug was fixed with version 3.2.1. The directions at https://packagecontrol.io/docs/troubleshooting should help you get running again.

@hobbyman There haven't been any changes to address the failure you are seeing. This is mostly because I don't know why OpenSSL things the certificate is invalid.

**Comment by manchicken** [comment]
Reinstalling didn't help.

```
Package Control: Error downloading package. HTTP exception InvalidCertificateException (Host codeload.github.com returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://codeload.github.com/babel/babel-sublime/zip/v8.6.3.
error: Package Control
```

I'm getting this for all packages.

**Comment by brikai** [comment]
Same here, really frustrating.  I've tried clearing out my keychain, purging and re-installing PC, purging and re-installing Sublime entirely... nothing works.  All happened after I upgraded to 10.12.  I should have just left it alone until a fix was discovered but now I'm left with a clean install of Sublime with no modules and no access to PC.

**Comment by drew-wallace** [comment]
I'm also on macOS 10.12, Sumblime Text 3 build 3126. I am getting this when selecting Install Package:

```
Exception in thread Thread-12:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "package_control.commands.install_package_command in /Users/drewwallace/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 45, in run
  File "package_control.package_installer in /Users/drewwallace/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 58, in make_package_list
  File "package_control.package_manager in /Users/drewwallace/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 687, in list_available_packages
  File "package_control.package_manager in /Users/drewwallace/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 551, in _list_available
  File "package_control.package_manager in /Users/drewwallace/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 414, in list_repositories
TypeError: 'NoneType' object is not subscriptable
```

**Comment by hobbyman** [comment]
I did wipe out all of the preferences and caches. Got a fresh copy of Sublime. That worked up until this week. I'm guessing our network nazis have changed something yet again.

I tried Atom and Visual Studio Code. Oh boy, just not the same. Both of those did have the option to NOT use SSL when installing packages / updates. Would love to see that implemented, @wbond

**Comment by manchicken** [comment]
I think it's part of the build. I switched to `curl` and it works just fine.

**Comment by hobbyman** [comment]
@manchicken, what do you mean by you "switched to curl"? 

Is that part of ST3 build 3126?
I don't see anything in the preferences that would clue me into using 'curl'

**Comment by hobbyman** [comment]
Sorry, @manchicken, I found that in the package control settings, works like a charm!!

**Comment by wgregp** [comment]
Same issue, but I managed to solve it. It appears that the default `Package Control.merged-ca-bundle` file (as installed today, 9 Nov 2016) does not include the `DigiCert High Assurance EV Root CA` certificate, which is what GitHub uses. I exported that certificate from Keychain Access as a `.pem` file, pasted the contents into `Package Control.user-ca-bundle`, restarted Sublime Text, and everything now seems to work.
